### PR TITLE
Parser: fix `a-b -c` incorrectly parsed as `a - (b - c)`

### PR DESCRIPTION
**Pull request by asterite**
Fixes https://forum.crystal-lang.org/t/results-of-a-b-c-and-a-b-c-different/2402

### CODE REVIEW

File: src/compiler/crystal/syntax/parser.cr [hunk 1]
```diff
@@ -4168,6 +4171,16 @@ module Crystal
       node
     end
 
+    def next_comes_plus_or_minus?
+      pos = current_pos
+      while current_char.ascii_whitespace?
+        next_char_no_column_increment
+      end
+      comes_plus_or_mius = current_char == '+' || current_char == '-'
```
Comment (vlazar) [inline src/compiler/crystal/syntax/parser.cr:4179]:
```suggestion
      comes_plus_or_minus = current_char == '+' || current_char == '-'
```

File: src/compiler/crystal/syntax/parser.cr [hunk 2]
```diff
@@ -4168,6 +4171,16 @@ module Crystal
...
+      self.current_pos = pos
+      comes_plus_or_mius
```
Comment (vlazar) [inline src/compiler/crystal/syntax/parser.cr:4181]:
```suggestion
      comes_plus_or_minus
```

File: spec/compiler/parser/parser_spec.cr [hunk 3]
```diff
@@ -111,6 +111,18 @@ module Crystal
     it_parses "1 / -2", Call.new(1.int32, "/", -2.int32)
     it_parses "2 / 3 + 4 / 5", Call.new(Call.new(2.int32, "/", 3.int32), "+", Call.new(4.int32, "/", 5.int32))
     it_parses "2 * (3 + 4)", Call.new(2.int32, "*", Expressions.new([Call.new(3.int32, "+", 4.int32)] of ASTNode))
+    it_parses "a = 1; b = 2; c = 3; a-b-c", Expressions.new([
+      Assign.new("a".var, 1.int32),
+      Assign.new("b".var, 2.int32),
+      Assign.new("c".var, 3.int32),
+      Call.new(Call.new("a".var, "-", "b".var), "-", "c".var),
+    ])
+    it_parses "a = 1; b = 2; c = 3; a-b -c", Expressions.new([
```
Comment (jhass) [inline spec/compiler/parser/parser_spec.cr:120]: I suppose you wanted to change this to `+`es :)

Comment (asterite) [reply spec/compiler/parser/parser_spec.cr:120]: No, the difference is there's a space

File: src/compiler/crystal/syntax/parser.cr [hunk 4]
```diff
@@ -4168,6 +4171,16 @@ module Crystal
       node
     end
 
+    def next_comes_plus_or_minus?
+      pos = current_pos
+      while current_char.ascii_whitespace?
+        next_char_no_column_increment
+      end
+      comes_plus_or_minus = current_char == '+' || current_char == '-'
```
Comment (Sija) [inline src/compiler/crystal/syntax/parser.cr:4179]:
```suggestion
      comes_plus_or_minus = current_char.in?('+', '-')
```

File: spec/compiler/parser/parser_spec.cr (hunk 3 above)
Comment (jhass) [reply spec/compiler/parser/parser_spec.cr:120]: Ah. How about specs for `+` then? :D

**Comment by asterite** [comment]
Closing because I'm not sure this is the best way to solve this.

**Comment by Sija** [comment]
@asterite Perhaps it's not _the best_ but it works, so why to throw the baby with the bathwater?

**Comment by asterite** [comment]
Nope

**Comment by asterite** [comment]
Feel free to open a PR with this diff, I won't babysit it.

**Comment by straight-shoota** [comment]
There should still be a bug report for this.

**Comment by asterite** [comment]
I think this one should be merged before 1.0

**Pull request merged**